tls/tls.md

```diff
@@ -122,6 +124,16 @@ How the public key is encoded into the `Data` bytes depends on the Key Type.
 - `Secp256k1`: Only the compressed form of the public key. 33 bytes.
 - The rest of the keys are encoded as a [SubjectPublicKeyInfo structure](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1) in PKIX, ASN.1 DER form.
 
+## ALPN
+
+"libp2p" is used as the application protocol for [ALPN](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation).
+
+The handshake MUST be aborted if both peers can't agree on the application protocol.
```

Comment:
This is underspecified. You should say: The server MUST abort if...

Reply:
How about:

> The server MUST abort the handshake if the client does not send a supported application protocol.

Reply:
Sounds a bit like you're blaming the client for misbehaving, why not the slightly less accusatory:
> The server MUST abort the handshake if it doesn't support any of the application protocols offered by the client.

Reply:
Well the client did send the wrong thing 😜 

The amendment sounds fine though.

```suggestion
The server MUST abort the handshake if it doesn't support any of the application protocols offered by the client.
```